src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionValidator.cs

```diff
@@ -117,6 +117,11 @@ private static SelectionInfo ApplySpecialCases(SelectionInfo selectionInfo, Sour
                 }
             }
 
+            if (_localFunction && selectionInfo.CommonRootFromOriginalSpan.AncestorsAndSelf().Any(c => c.IsKind(SyntaxKind.BaseConstructorInitializer)))
```

Comment:
generally, walking up arbitrary heights is not good.  because it breaks legal cases (like a lambda in the this/base which might have code to extract into a local function).  so try to walk up and see if you have some AnonmousMethodSyntax above  you, and allow extraction then (we may have helpers for this, not sure).

Reply:
Ah I see... didn't think about anonymouse method within. Lambdas 🤦